adapter/src/hermes/adapter/interceptor.cc

```diff
@@ -0,0 +1,94 @@
+/**
+ * Internal headers
+ */
+#include "interceptor.h"
+
+/**
+ * Define Interceptor list for adapter.
+ */
+hermes::adapter::InterceptorList list;
```

Comment:
`hermes_stdio_adapter_test` is failing on my system (Ubuntu 20.04). Before `main`, ` libnuma.so.1` attempts to call `fopen` on `/proc/self/status` which the adapter intercepts. However, `IsTracked()` returns `true` for this file because `hermes::adapter::InterceptorList list` has been zero-initialized by the compiler (because it's storage duration is static) but the default constructor has not been called yet (i.e., its `path_exclusions` member is empty). So, the adapter attempts to buffer the file but Hermes initialization fails because MPI isn't initialized yet. I guess we need to create a default `InterceptorList` on the stack if the static `list` isn't initialized yet (which we can presumably detect if `path_exclusions` is empty, though I'm not sure how safe it is to access `std::` data structures that are zero-initialized). Thoughts?

Reply:
> though I'm not sure how safe it is to access std:: data structures that are zero-initialized

According to the "Access Outside of Lifetime" section [here](https://en.cppreference.com/w/cpp/language/lifetime), it's undefined behavior.

Reply:
Ok, what about moving path exclusion and inclusions as constant in interceptor.h and then using them as lists directly. We support user_path exclusions anyway which can be used by users.

Reply:
Still the same problem. `kPathExclusions` is empty the first time `IsTracked` is called. Can you somehow eliminate the global `list`?

Reply:
It works if you switch `kPathExclusions` from a vector to a `char*` array:
```c++
const char *kPathExclusions[] = {
    "/bin/",  "/boot/", "/dev/", "/etc/", "/lib/", "/opt/",
    "/proc/", "/sbin/", "/sys/", "/usr/", "/var/"
};
```
Although we're still relying on undefined behavior because we call `list.push_back` before its constructor is called. Best would be to eliminate the global.

Reply:
Also, can you please add `"/run/"` to the exclusions list?

Reply:
So you want me to switch to ```const char *kPathExclusions[]``` and add /run/. Is that correct?

Reply:
Also to remove the global list, I can create a singleton. I have tested it and commit the changes. Let me know if that approach works for you.

Reply:
The singleton seems to work. As a side note, initializing and finalizing Hermes several thousand times in a single process has reminded me that I never close the buffering files. I can do that in a separate PR though.

Reply:
Ok sounds good